pkg/apis/provisioning/v0alpha1/jobs.go

```diff
@@ -143,6 +149,20 @@ type DeleteJobOptions struct {
 	Paths []string `json:"paths,omitempty"`
 }
 
+type MoveJobOptions struct {
+	// Ref to the branch or commit hash to delete from
+	Ref string `json:"ref,omitempty"`
+	// Paths to be deleted. Examples:
+	// - dashboard.json (for a file)
+	// - a/b/c/other-dashboard.json (for a file)
+	// - nested/deep/ (for a directory)
+	// FIXME: we should validate this in admission hooks
```

Comment:
IIUC, this should fail if:
* unsafe values
* target is a file and source has a folder?

Reply:
noted, we will discuss those in the admissions hooks ticket :) 

https://github.com/grafana/git-ui-sync-project/issues/373